Building using cmake fails when linking due to unrecognized option 'ck_size'

Hello,

I'm trying to package Bit Magic for guix and use it as a dependency for one of my other projects.  The in-place build approach seems to work, but the cmake one appears broken with the following output.  I would like to use cmake as it is easier to use (from what I know) than using the more custom primary approach.

```
BitMagic$ mkdir bld
BitMagic$ cd bld/
BitMagic/bld$ cmake ..
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/anadon/Documents/code/guix-packages/BitMagic/bld
anadon@goodadvicemallard:~/Documents/code/guix-packages/BitMagic/bld$ ls
CMakeCache.txt  CMakeFiles  cmake_install.cmake  Makefile
anadon@goodadvicemallard:~/Documents/code/guix-packages/BitMagic/bld$ make
/gnu/store/2hxfv110z443qbzal6rwsny4sx13dys3-cmake-3.15.5/bin/cmake -S/home/anadon/Documents/code/guix-packages/BitMagic -B/home/anadon/Documents/code/guix-packages/BitMagic/bld --check-build-system CMakeFiles/Makefile.cmake 0
/gnu/store/2hxfv110z443qbzal6rwsny4sx13dys3-cmake-3.15.5/bin/cmake -E cmake_progress_start /home/anadon/Documents/code/guix-packages/BitMagic/bld/CMakeFiles /home/anadon/Documents/code/guix-packages/BitMagic/bld/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
make -f CMakeFiles/strsvsample03.dir/build.make CMakeFiles/strsvsample03.dir/depend
make[2]: Entering directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
cd /home/anadon/Documents/code/guix-packages/BitMagic/bld && /gnu/store/2hxfv110z443qbzal6rwsny4sx13dys3-cmake-3.15.5/bin/cmake -E cmake_depends "Unix Makefiles" /home/anadon/Documents/code/guix-packages/BitMagic /home/anadon/Documents/code/guix-packages/BitMagic /home/anadon/Documents/code/guix-packages/BitMagic/bld /home/anadon/Documents/code/guix-packages/BitMagic/bld /home/anadon/Documents/code/guix-packages/BitMagic/bld/CMakeFiles/strsvsample03.dir/DependInfo.cmake --color=
Scanning dependencies of target strsvsample03
make[2]: Leaving directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
make -f CMakeFiles/strsvsample03.dir/build.make CMakeFiles/strsvsample03.dir/build
make[2]: Entering directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
[  1%] Building CXX object CMakeFiles/strsvsample03.dir/samples/strsvsample03/strsvsample03.cpp.o
/usr/bin/c++   -I/home/anadon/Documents/code/guix-packages/BitMagic/src  -Wall -Wextra -Wno-ignored-qualifiers -march=core2 -fPIC -march=core2   -std=gnu++11 -o CMakeFiles/strsvsample03.dir/samples/strsvsample03/strsvsample03.cpp.o -c /home/anadon/Documents/code/guix-packages/BitMagic/samples/strsvsample03/strsvsample03.cpp
In file included from /home/anadon/Documents/code/guix-packages/BitMagic/src/bmserial.h:54,
                 from /home/anadon/Documents/code/guix-packages/BitMagic/src/bmsparsevec_serial.h:33,
                 from /home/anadon/Documents/code/guix-packages/BitMagic/samples/strsvsample03/strsvsample03.cpp:43:
/home/anadon/Documents/code/guix-packages/BitMagic/src/bmxor.h: In function ‘bm::id64_t bm::compute_xor_complexity_descr(const word_t*, const word_t*, bm::block_waves_xor_descr&, unsigned int&)’:
/home/anadon/Documents/code/guix-packages/BitMagic/src/bmxor.h:164:56: warning: comparison is always true due to limited range of data type [-Wtype-limits]
  164 |         if ((xor_change <= 1) && (x_descr.sb_change[i] >= 0))
      |                                   ~~~~~~~~~~~~~~~~~~~~~^~~~
[  2%] Linking CXX executable ../build/bin/strsvsample03
/gnu/store/2hxfv110z443qbzal6rwsny4sx13dys3-cmake-3.15.5/bin/cmake -E cmake_link_script CMakeFiles/strsvsample03.dir/link.txt --verbose=1
/usr/bin/c++   -Wall -Wextra -Wno-ignored-qualifiers -march=core2 -fPIC -march=core2   -Wl,-stack_size,0x100000000 -rdynamic CMakeFiles/strsvsample03.dir/samples/strsvsample03/strsvsample03.cpp.o  -o ../build/bin/strsvsample03 
ld: unrecognized -a option `ck_size'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/strsvsample03.dir/build.make:87: ../build/bin/strsvsample03] Error 1
make[2]: Leaving directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
make[1]: *** [CMakeFiles/Makefile2:130: CMakeFiles/strsvsample03.dir/all] Error 2
make[1]: Leaving directory '/home/anadon/Documents/code/guix-packages/BitMagic/bld'
make: *** [Makefile:87: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ set( CMAKE_VERBOSE_MAKEFILE on )
 
 # SET(BMOPTFLAGS "BMSSE42OPT" CACHE STRING "User-specified optimization flag")
 
+find_package(Threads REQUIRED)
+
 set(CMAKE_BINARY_DIR ${CMAKE_SOURCE_DIR}/build/bin)
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR})
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR})
@@ -28,7 +30,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR
 
     set(flags "-Wall -Wextra -Wno-ignored-qualifiers -march=core2 -fPIC")
     set(optflags -g0 -O3)
-    SET( CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -Wl,-stack_size,0x100000000")
+    SET( CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -Wl,-lpthread")
 
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
     set(flags, "-tpp7 -march=core2 -restrict -DBM_HASRESTRICT -fno-fnalias -Wall")
@@ -43,11 +45,11 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 
     set(flags "/W4 /EHsc /F 5000000 ")
     set(optflags "-O2")
-    SET( CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} /STACK:\"10000000\"")
+    SET( CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -lpthread ")
 endif()
 
-set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} ${flags} ${bmoptf}")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flags} ${bmoptf}")
+set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} ${flags} ${bmoptf} -pthread")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flags} ${bmoptf} -pthread")
 
 add_executable(bmsvutil ${PROJECT_SOURCE_DIR}/utils/svutil/svutil.cpp)
 

